### ISSUE TITLE
AZD hanging while waiting for user to input mandatory parameters without default values

### ISSUE THREAD

**Issue by yorek** [Azure/azure-dev#1176]
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
azd version 0.4.0-beta.1 (commit 2bf7a529914d86d8e9f5d7b6f95a4f20a2351e01)

**Describe the bug**
azd is not asking for value of mandatory parameters, for which the value has not been provided in the mail.parameters.json

**To Reproduce**
Define a parameter without a default value, for example, in the bicep script define something like

```
param foo string
```

then run `azd provision`. The execution will hang at 

```
Compiling Bicep template... |
```

if you cancel the execution, you'll get this error:

```
Error: planning deployment: planning infrastructure provisioning: prompting for deployment parameter: interrupt
```

which seems to indicate that azd was waiting for some input, but I got nothing in the console. I'm using WSL.

**Expected behavior**
Ask for the value of mandatory parameters without default values

**Environment**
WSL Ubuntu

**Additional context**
What I really need to do, is to initialize the .env file by asking some additional values to the end user. Aside from the location and subscription id, it would be great if azd could use the description attribute defined in the .bicep file, so that it will be presented to the user during initialization and (optionally) saved in the .env file.

At the end of `azd init` I should have a file like the following:

```
AZURE_ENV_NAME="xyz"
AZURE_LOCATION="eastus2"
AZURE_PRINCIPAL_ID="..."
AZURE_SUBSCRIPTION_ID="..."
MY_PARAM_1=something
MY_PARAM_2=something
```

if in the '' file, I have configured the param like:

```
 "my_param_1": {
      "value": "${MY_PARAM_1}",
    }
```

**Comment by weikanglim** [comment]
@ellismg I believe this was covered in your PoC for environment redesign? Does this fit under #808

**Comment by yorek** [comment]
Thanks! would this be available in the next version (0.5 I guess)? Do you have an ETA for it?

**Comment by rajeshkamal5050** [comment]
> Thanks! would this be available in the next version (0.5 I guess)? Do you have an ETA for it?

Our next major release is targeted for Jan 11 - https://github.com/Azure/azure-dev/milestone/8
